### ISSUE TITLE
Unable to build 1.2.18 (`undefined reference to `lsp::ui::Factory::pRoot`)

### ISSUE THREAD

**Issue by dvzrv** [lsp-plugins/lsp-plugins#449]
Hi! :wave: 

When trying to build 1.2.18 for Arch Linux I ran into the following issue:

```
respack lsp-plugin-fw-res.cpp
Packing resources to /build/lsp-plugins/src/lsp-plugins-src-1.2.18/.build/target/lsp-plugin-fw/lsp-plugin-fw-res.cpp from /build/lsp-plugins/src/lsp-plugins-src-1.2.18/.build/target/lsp-plugin-fw/res
  found file: manifest.json
Execution time: 0.01 s
  g++  [lsp-plugin-fw] validator
  g++  [lsp-plugin-fw] lv2ttl_gen
  g++  [lsp-plugin-fw] lsp-plugins-lv2.so
../../lsp-plugins-compressor/src/main/plug/compressor.cpp:39:24: warning: type ‘struct plugin_settings_t’ violates the C++ One Definition Rule [-Wodr]
   39 |         typedef struct plugin_settings_t
      |                        ^
../../lsp-plugins-crossover/src/main/plug/crossover.cpp:41:24: note: a different type is defined in another translation unit
   41 |         typedef struct plugin_settings_t
      |                        ^
../../lsp-plugins-compressor/src/main/plug/compressor.cpp:42:37: note: the first difference of corresponding definitions is field ‘sc’
   42 |             bool                    sc;
      |                                     ^
../../lsp-plugins-crossover/src/main/plug/crossover.cpp:44:37: note: a field with different name is defined in another translation unit
   44 |             uint8_t                 mode;
      |                                     ^
../../lsp-plugins-compressor/src/main/plug/compressor.cpp:39:24: warning: type ‘struct plugin_settings_t’ violates the C++ One Definition Rule [-Wodr]
   39 |         typedef struct plugin_settings_t
      |                        ^
../../lsp-plugins-crossover/src/main/plug/crossover.cpp:41:24: note: a different type is defined in another translation unit
   41 |         typedef struct plugin_settings_t
      |                        ^
../../lsp-plugins-compressor/src/main/plug/compressor.cpp:42:37: note: the first difference of corresponding definitions is field ‘sc’
   42 |             bool                    sc;
      |                                     ^
../../lsp-plugins-crossover/src/main/plug/crossover.cpp:44:37: note: a field with different name is defined in another translation unit
   44 |             uint8_t                 mode;
      |                                     ^
../../lsp-plugins-compressor/src/main/plug/compressor.cpp:39:24: warning: type ‘struct plugin_settings_t’ violates the C++ One Definition Rule [-Wodr]
   39 |         typedef struct plugin_settings_t
      |                        ^
../../lsp-plugins-crossover/src/main/plug/crossover.cpp:41:24: note: a different type is defined in another translation unit
   41 |         typedef struct plugin_settings_t
      |                        ^
../../lsp-plugins-compressor/src/main/plug/compressor.cpp:42:37: note: the first difference of corresponding definitions is field ‘sc’
   42 |             bool                    sc;
      |                                     ^
../../lsp-plugins-crossover/src/main/plug/crossover.cpp:44:37: note: a field with different name is defined in another translation unit
   44 |             uint8_t                 mode;
      |                                     ^
../../lsp-runtime-lib/src/main/fmt/lspc/AudioWriter.cpp: In member function ‘parse_parameters’:
../../lsp-runtime-lib/src/main/fmt/lspc/AudioWriter.cpp:321:68: warning: argument 1 value ‘18446744073709551615’ exceeds maximum object size 9223372036854775807 [-Walloc-size-larger-than=]
  321 |             pBuffer         = new float[p->channels * BUFFER_FRAMES];
      |                                                                    ^
../../lsp-runtime-lib/src/main/fmt/lspc/AudioWriter.cpp:321:68: note: in a call to built-in allocation function ‘operator new []’
../../lsp-runtime-lib/src/main/fmt/lspc/AudioWriter.cpp:321:68: warning: argument 1 value ‘18446744073709551615’ exceeds maximum object size 9223372036854775807 [-Walloc-size-larger-than=]
../../lsp-runtime-lib/src/main/fmt/lspc/AudioWriter.cpp:321:68: note: in a call to built-in allocation function ‘operator new []’
../../lsp-runtime-lib/src/main/fmt/lspc/AudioWriter.cpp: In member function ‘parse_parameters’:
../../lsp-runtime-lib/src/main/fmt/lspc/AudioWriter.cpp:321:68: warning: argument 1 value ‘18446744073709551615’ exceeds maximum object size 9223372036854775807 [-Walloc-size-larger-than=]
  321 |             pBuffer         = new float[p->channels * BUFFER_FRAMES];
      |                                                                    ^
../../lsp-runtime-lib/src/main/fmt/lspc/AudioWriter.cpp:321:68: note: in a call to built-in allocation function ‘operator new []’
../../lsp-runtime-lib/src/main/fmt/lspc/AudioWriter.cpp:321:68: warning: argument 1 value ‘18446744073709551615’ exceeds maximum object size 9223372036854775807 [-Walloc-size-larger-than=]
../../lsp-runtime-lib/src/main/fmt/lspc/AudioWriter.cpp:321:68: note: in a call to built-in allocation function ‘operator new []’
../../lsp-runtime-lib/src/main/fmt/lspc/AudioWriter.cpp: In member function ‘parse_parameters’:
../../lsp-runtime-lib/src/main/fmt/lspc/AudioWriter.cpp:321:68: warning: argument 1 value ‘18446744073709551615’ exceeds maximum object size 9223372036854775807 [-Walloc-size-larger-than=]
  321 |             pBuffer         = new float[p->channels * BUFFER_FRAMES];
      |                                                                    ^
../../lsp-runtime-lib/src/main/fmt/lspc/AudioWriter.cpp:321:68: note: in a call to built-in allocation function ‘operator new []’
../../lsp-runtime-lib/src/main/fmt/lspc/AudioWriter.cpp:321:68: warning: argument 1 value ‘18446744073709551615’ exceeds maximum object size 9223372036854775807 [-Walloc-size-larger-than=]
../../lsp-runtime-lib/src/main/fmt/lspc/AudioWriter.cpp:321:68: note: in a call to built-in allocation function ‘operator new []’
/usr/bin/ld: /tmp/ccgtFbuu.ltrans25.ltrans.o: in function `lsp::validator::main(int, char const**) [clone .isra.0]':
/usr/src/debug/lsp-plugins/lsp-plugins-src-1.2.18/modules/lsp-plugin-fw/include/lsp-plug.in/plug-fw/ui/Factory.h:73:(.text._ZN3lsp9validator4mainEiPPKc.isra.0+0x143c): undefined reference to `lsp::ui::Factory::pRoot'
/usr/bin/ld: /build/lsp-plugins/src/lsp-plugins-src-1.2.18/.build/target/lsp-plugin-fw/utils-bin/validator: hidden symbol `_ZN3lsp2ui7Factory5pRootE' isn't defined
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:1249: /build/lsp-plugins/src/lsp-plugins-src-1.2.18/.build/target/lsp-plugin-fw/utils-bin/validator] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/build/lsp-plugins/src/lsp-plugins-src-1.2.18/modules/lsp-plugin-fw/src'
make[2]: *** [Makefile:67: install] Error 2
make[2]: Leaving directory '/build/lsp-plugins/src/lsp-plugins-src-1.2.18/modules/lsp-plugin-fw'
make[1]: *** [Makefile:145: install] Error 2
make[1]: Leaving directory '/build/lsp-plugins/src/lsp-plugins-src-1.2.18/src'
make: *** [Makefile:68: install] Error 2
make: Leaving directory '/build/lsp-plugins/src/lsp-plugins-src-1.2.18'
```

I also tried building without LTO, but that results in a similar failure:
```
respack lsp-plugin-fw-res.cpp
Packing resources to /build/lsp-plugins/src/lsp-plugins-src-1.2.18/.build/target/lsp-plugin-fw/lsp-plugin-fw-res.cpp from /build/lsp-plugins/src/lsp-plugins-src-1.2.18/.build/target/lsp-plugin-fw/res
  found file: manifest.json
Execution time: 0.01 s
  g++  [lsp-plugin-fw] validator
  g++  [lsp-plugin-fw] lv2ttl_gen
  g++  [lsp-plugin-fw] lsp-plugins-lv2.so
/usr/bin/ld: /build/lsp-plugins/src/lsp-plugins-src-1.2.18/.build/target/lsp-plugin-fw/util/validator/common.o: in function `lsp::ui::Factory::root()':
/usr/src/debug/lsp-plugins/lsp-plugins-src-1.2.18/modules/lsp-plugin-fw/include/lsp-plug.in/plug-fw/ui/Factory.h:73:(.text._ZN3lsp9validator7find_uiEPKNS_4meta8plugin_tE+0x12): undefined reference to `lsp::ui::Factory::pRoot'
/usr/bin/ld: /build/lsp-plugins/src/lsp-plugins-src-1.2.18/.build/target/lsp-plugin-fw/utils-bin/validator: hidden symbol `_ZN3lsp2ui7Factory5pRootE' isn't defined
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:1249: /build/lsp-plugins/src/lsp-plugins-src-1.2.18/.build/target/lsp-plugin-fw/utils-bin/validator] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/build/lsp-plugins/src/lsp-plugins-src-1.2.18/modules/lsp-plugin-fw/src'
make[2]: *** [Makefile:67: install] Error 2
make[2]: Leaving directory '/build/lsp-plugins/src/lsp-plugins-src-1.2.18/modules/lsp-plugin-fw'
make[1]: *** [Makefile:145: install] Error 2
make[1]: Leaving directory '/build/lsp-plugins/src/lsp-plugins-src-1.2.18/src'
make: *** [Makefile:68: install] Error 2
make: Leaving directory '/build/lsp-plugins/src/lsp-plugins-src-1.2.18'
```

**Comment by sadko4u** [comment]
Hello! Very strange. What's your compiler version?

**Comment by dvzrv** [comment]
We're on gcc 14.2.1.

Here are also the full check logs:
[lsp-plugins-1.2.18-1-x86_64-check.log](https://github.com/user-attachments/files/17307938/lsp-plugins-1.2.18-1-x86_64-check.log)

**Comment by sadko4u** [comment]
The logs are not full. It's very strange because the CI builds plugins very well on Arch Linux:
https://github.com/lsp-plugins/lsp-plugins/actions/runs/11226506229

**Comment by sadko4u** [comment]
Can you give more information about what arguments you're passing to `make` calls?

**Comment by dvzrv** [comment]
Sorry, I should have added that this happens in test (after successful build), more specifically when attempting a local installation of the lv2 plugins so they can be tested with lv2lint:
https://gitlab.archlinux.org/archlinux/packaging/packages/lsp-plugins/-/blob/6022e7b9f318cba6de454e70e1010a5db6cf3299/PKGBUILD#L66

The build runs fine (the previous log file is from the check function), as can be seen in the full build log: 
[lsp-plugins-1.2.18-1-x86_64-build.log](https://github.com/user-attachments/files/17316521/lsp-plugins-1.2.18-1-x86_64-build.log)

**Comment by sadko4u** [comment]
Oh... the `make FEATURES=lv2` should be replaced with `make INSTALL_FEATURES=lv2` now. Sorry if I didn't mention that in CHANGELOG.

**Comment by sadko4u** [comment]
Anyway, I think you should not hurry by changing your build script. I think I can update make files to keep `FEATURES` parameter instead of `INSTALL_FEATURES`.

**Comment by sadko4u** [comment]
@dvzrv seems that I've fixed the issue and you do not need to change your test scripts because I've returned the `FEATURES` variable back.
In the upcoming 1.2.19 release all should work. And I'll switch back to tar.gz archive for source code.
It also could be nice if you could try to build from the `devel` branch to check that all is OK for you.

**Comment by sadko4u** [comment]
@dvzrv Fixed build in 1.2.19.
